packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -366,22 +359,7 @@ function DashboardLayout(props: DashboardLayoutProps) {
                   </Box>
                 </React.Fragment>
               ) : null}
-              <Link href="/" style={{ color: 'inherit', textDecoration: 'none' }}>
-                <Stack direction="row" alignItems="center">
-                  <LogoContainer>{branding?.logo ?? <ToolpadLogo size={40} />}</LogoContainer>
-                  <Typography
-                    variant="h6"
-                    sx={{
-                      color: (theme.vars ?? theme).palette.primary.main,
-                      fontWeight: '700',
-                      ml: 1,
-                      whiteSpace: 'nowrap',
-                    }}
-                  >
-                    {applicationTitle}
-                  </Typography>
-                </Stack>
-              </Link>
+              {slots?.appTitle ? <slots.appTitle /> : <AppTitleComponent {...appTitleProps} />}
```

Comment:
> Should people be able to pass in `slotProps` of the type `Branding` to the custom slot if they set it? Then it would match the props of `AppTitleComponent`, right? 

Agreed, missed this

> Also I'd just have a single branding prop for these slot components instead of all the props inside Branding, then maybe we could add other stuff later.

Makes sense, so merge all the `Branding` properties inside a single `branding` prop for the `AppTitle`

For the slot type definition, if I used 

```tsx
appTitle?: React.JSXElementConstructor<AppTitleProps>;
```

that would cause type errors if someone wanted to create a custom element with their own props for this component, right? Would it be better to use `React.ElementType` instead here?

Reply:
> that would cause type errors if someone wanted to create a custom element with their own props for this component, right? Would it be better to use React.ElementType instead here?

Could we make it so that the custom component automatically has these default props but can be expanded with whatever other properties people want to add?

But if not and that causes errors, we haven't really decided on a pattern for the slots yet so if using `React.ElementType` fixes that we can go with that pattern.

In this particular case you don't even need to pass branding as a prop to the slot if that ends up solving this problem, those branding values can just be read from the context if that's the best solution.

Reply:
I guess going with `React.ElementType` like you did should work if it shows no errors.